File: 349_Intersection_of_Two_Arrays.js
var intersection = function(nums1, nums2) {
    let ans = []
    for(let i = 0; i < nums2.length; i++){
        let index = nums1.indexOf(nums2[i])
        if(index != -1){
            if(ans.indexOf(nums2[i]) == -1){
                ans.push(nums2[i])
            }
            nums2[i] = 1001
            nums1[index] = 1001
        }
    }
    return ans
};

File: 349_Intersection_of_Two_Arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        unordered_map<int,int> mp1;
        unordered_map<int,int> mp2;
        for(int x: nums1){
            mp1[x]++;
        }
        vector<int> ans;
        for(int x : nums2){
            if(mp1.find(x) != mp1.end() && mp2.find(x) == mp2.end()){
                ans.push_back(x);
                mp2[x]++;
            }
        }
        return ans;
    }
};
